pkgs/tools/networking/wireguard-go/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, buildGoPackage, fetchurl }:
+
+buildGoPackage rec {
+  name = "wireguard-go-${version}";
+  version = "0.0.20180514";
+
+  goPackagePath = "wireguard-go";
+
+  src = fetchurl {
+    url = "https://git.zx2c4.com/wireguard-go/snapshot/wireguard-go-${version}.tar.xz";
+    sha256 = "1bn49a67m2ab0l9lq3zh2mfbbppmyg34klqi3069sjn6lg2hlajs";
+  };
+
+  goDeps = ./deps.nix;
+
+  patchPhase = ''
```

Comment:
Better use `postPatch` instead of `patchPhase` so that one can still use `patches = [ ... ]` to apply custom patches if they want.

Reply:
✅